sunkit_pyvista/plotter.py

```diff
@@ -1,10 +1,138 @@
+import functools
+
+import numpy as np
+import pyvista as pv
+
+import astropy.units as u
+from astropy.constants import R_sun
+from astropy.coordinates import SkyCoord
+
+from sunpy.map.maputils import all_corner_coords_from_map
+from sunkit_pyvista.draw import Draw
+
 
 __all__ = ['SunpyPlotter']
 
 
 class SunpyPlotter:
     """
     A plotter for 3D data.
-
     This class wraps `pyvsita.Plotter` to provide coordinate-aware plotting.
+    For now, all coordinates are converted to
+    and distance units are such that :math:`R_{sun} = 1`.
+
+    Parameters
+    ----------
+    coordinate_frame : astropy.coordinates.BaseFrame
+        Coordinate frame of the plot. The x, y, z axes of the pyvista plotter
+        will be the x, y, z axes in this coordinate system.
     """
+
+    def __init__(self, coordinate_frame=None):
+        self._plotter = pv.Plotter()
+        self.draw = Draw(coordinate_frame)
+
+    @property
+    def plotter(self):
+        """
+        `pyvista.Plotter`.
+        """
+        return self._plotter
+
+    @functools.wraps(pv.Plotter.show)
+    def show(self, *args, **kwargs):
+        """
+        Show the plot.
+        """
+        self.plotter.show(*args, **kwargs)
+
+    def _pyvista_mesh(self, m):
```

Comment:
I think this should live in a separate file as a function (not a method) that takes the signature
```python
def pyvista_mesh(smap, coordinate_frame):
```

Reply:
Okay, sure. I'll restructure this PR with the initial code that you wrote and make another one with the splitting of the code into separate classes and files.